File: Golang/171. Excel Sheet Column Number.go
func titleToNumber(s string) int {
    length := len(s)
    sum := 0
    factor := 1
    for i := length-1; i >= 0; i-- {
        sum += (int(s[i] - 'A') + 1) * factor
        factor *= 26
    }
    return sum
}

File: Python3/171. Excel Sheet Column Number.py
class Solution:
    def titleToNumber(self, s):
        """
        :type s: str
        :rtype: int
        """
        chardict = dict(zip(string.ascii_uppercase, range(1, 27)))
        sum = 0
        factor = 1
        for char in s[::-1]:
            sum += chardict[char] * factor
            factor *= 26
        return sum
